activity_browser/ui/tables/models/base.py

```diff
@@ -54,6 +54,7 @@ def data(self, index, role=Qt.DisplayRole):
 
```

Comment:
This method has become very unclear over the last commits. Maybe we should chop it up into different functions/methods. e.g. a value function that returns the value.

Reply:
A switch case block also comes to mind. Those are awfully quick, meant for enums like these, and easy to read. Just spitballing

Reply:
Maybe the whole date thing could go in a child class for the DB table as well

Reply:
> This method has become very unclear over the last commits. Maybe we should chop it up into different functions/methods. e.g. a value function that returns the value.

Agree, it's a bit messy atm

> A switch case block also comes to mind. Those are awfully quick, meant for enums like these, and easy to read. Just spitballing

`switch` only exists from `py=3.10`, so we can't implement this until we drop support for `3.9`. Though `elif` is essentially a switch, and I don't think the speed matters in this case.
 
> Maybe the whole date thing could go in a child class for the DB table as well

Also agree